src/coffea/analysis_tools.py

```diff
@@ -488,15 +1005,13 @@ def __add_eager(self, name, selection, fill_value):
             raise ValueError(f"Expected a boolean array, received {selection.dtype}")
         if len(self._names) == 0:
             self._data = numpy.zeros(len(selection), dtype=self._dtype)
-        if isinstance(selection, numpy.ndarray) and not isinstance(
-            self._data, numpy.ndarray
-        ):
+        if isinstance(selection, numpy.ndarray) and self.delayed_mode:
             raise ValueError(
-                f"New selection '{name}' is not eager while PackedSelection is!"
+                f"New selection '{name}' is not delayed while PackedSelection is!"
             )
         elif len(self._names) == self.maxitems:
             raise RuntimeError(
-                f"Exhausted all slots in {self}, consider a larger dtype or fewer selections"
+                "Exhausted all slots in this PackedSelection, consider a larger dtype or fewer selections"
```

Comment:
`"Exhausted all slots in PackedSelection: {self}, consider a larger dtype or fewer selections"`

This is more clear in the case you have multiple packed selections floating around.